guiclient/salesOrder.cpp

```
@@ -1393,7 +1395,20 @@ void salesOrder::sPopulateMenu(QMenu *pMenu)
           menuItem->setEnabled(_privileges->check("FirmSalesOrder"));
         }
         pMenu->addAction(tr("Close Line..."), this, SLOT(sAction()));
-        pMenu->addAction(tr("Delete Line..."), this, SLOT(sDelete()));
+        pMenu->addAction(tr("Cancel Line..."), this, SLOT(sCancel()));
+
+        XSqlQuery invhist;
+        invhist.prepare("SELECT 1 "
+                        "  FROM invhist "
+                        " WHERE invhist_ordnumber=formatSoNumber(:soitemid) "
+                        "   AND invhist_ordtype='SO';");
+        invhist.bindValue(":soitemid", _soitem->id());
+        invhist.exec();
+        if (!invhist.first())
+          pMenu->addAction(tr("Delete Line..."), this, SLOT(sDelete()));
+        else if(ErrorReporter::error(QtCriticalMsg, this, "Error checking invhist",
```

Errors will be masked by the `! first()` call above.